synapse/storage/background_updates.py

```diff
@@ -133,10 +133,12 @@ def do_next_background_update(self, desired_duration_ms):
             updates = yield self._simple_select_list(
                 "background_updates",
                 keyvalues=None,
-                retcols=("update_name",),
+                retcols=("update_name", "depends_on"),
             )
+            in_flight = set(update["update_name"] for update in updates)
             for update in updates:
-                self._background_update_queue.append(update['update_name'])
+                if update["depends_on"] not in in_flight:
+                    self._background_update_queue.append(update['update_name'])
```

Comment:
Oh wait this will probably just work when ``do_next_background_update`` is called once the dependency finishes.